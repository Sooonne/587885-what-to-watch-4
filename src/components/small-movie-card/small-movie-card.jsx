import React, {PureComponent} from 'react';
import propTypes from "prop-types";
import VideoPlayer from '../video-player/video-player.jsx';
import DEFAULT_PROPTYPES from "../../prop-type-units/prop-types-units.js";
import {withRouter, Link} from 'react-router-dom';

export class SmallMovieCard extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isPlaying: false,
    };

    this._handleMovieClick = this._handleMovieClick.bind(this);
  }

  _handleMovieClick() {
    const {movie, history} = this.props;
    history.push(`/films/${movie.id}`);
  }


  render() {
    const {movie} = this.props;
    return (
      <article className="small-movie-card catalog_movies-card"
        onMouseOver={() => {
          this.setState({
            isPlaying: true,
          });
        }}
        onMouseOut={() => {
          this.setState({
            isPlaying: false,
          });
        }}
        onClick = {this._handleMovieClick}
      >
        <div className="small-movie-card__image">
          <VideoPlayer
            isPlaying={this.state.isPlaying}
            src={movie.src}
            poster={movie.poster}
          />
        </div>
        <h3 className="small-movie-card__title">
          <Link to={`/films/${movie.id}`} className="small-movie-card__link">{movie.title}</Link>
        </h3>
      </article>
    );
  }
}

SmallMovieCard.propTypes = {
  movie: DEFAULT_PROPTYPES.MOVIE_CARD,
  history: propTypes.object.isRequired
};


export default withRouter(SmallMovieCard);
