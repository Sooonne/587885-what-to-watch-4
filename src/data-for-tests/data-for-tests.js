export const PROMO_FILM = {
  title: `Star Wars`,
  genre: `Drama`,
  release: 2017,
  img: `img/the-grand-budapest-hotel.jpg`,
  bg: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 8.9,
  ratingCount: 240,
  id: 9,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

export const MOVIES = [
  {
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    genre: `Drama`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 1,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 10,
    isFaborite: false,
  },
  {
    title: `Bohemian Rhapsody`,
    genre: `Documentary`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 2,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 20,
    isFaborite: false,
  },
  {
    title: `Macbeth`,
    genre: `Comedy`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 3,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 0,
    isFaborite: false,
  },
  {
    title: `Aviator`,
    genre: `Comedy`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 4,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 40,
    isFaborite: false,
  },
  {
    title: `We need to talk about Kevin`,
    genre: `Thriller`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 5,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 50,
    isFaborite: false,
  },
  {
    title: `What We Do in the Shadows`,
    genre: `Thriller`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 6,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 60,
    isFaborite: false,
  },
  {
    title: `Pulp Fiction`,
    genre: `Crime`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 7,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 70,
    isFaborite: false,
  },
  {
    title: `Snatch`,
    genre: `Crime`,
    release: 2017,
    img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    bg: `img/bg-the-grand-budapest-hotel.jpg`,
    poster: `img/the-grand-budapest-hotel-poster.jpg`,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    ratingScore: 8.9,
    ratingCount: 240,
    id: 8,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
    bgColor: `#ffffff`,
    runTime: 80,
    isFaborite: false,
  }
];


export const MOVIE = {
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  genre: `Drama`,
  release: 2017,
  img: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  bg: `img/bg-the-grand-budapest-hotel.jpg`,
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  ratingScore: 8.9,
  ratingCount: 240,
  id: 1,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};

export const VIDEO = {
  poster: `img/the-grand-budapest-hotel-poster.jpg`,
  src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
};


export const REVIEW = {
  id: 1,
  user: {
    id: 2,
    name: `Kate Muir`,
  },
  rating: 8.9,
  comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
  date: `2019-05-08T14:13:56.569Z`
};

export const REVIEWS = [
  {
    id: 1,
    user: {
      id: 2,
      name: `Kate Muir`,
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 2,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 3,
    user: {
      id: 8,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 4,
    user: {
      id: 1,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
  {
    id: 5,
    user: {
      id: 4,
      name: `Kate Muir`
    },
    rating: 8.9,
    comment: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
    date: `2019-05-08T14:13:56.569Z`
  },
];
