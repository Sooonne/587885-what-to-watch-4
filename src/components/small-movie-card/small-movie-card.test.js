import React from "react";
import renderer from "react-test-renderer";
import {SmallMovieCard} from "./small-movie-card.jsx";
import {MOVIE} from "../../data-for-tests/data-for-tests";

import {createMemoryHistory} from 'history';
const history = createMemoryHistory();

it(`Should SmallMovieCard render correctly`, () => {
  const tree = renderer
    .create(<SmallMovieCard
      movie = {MOVIE}
      onMovieCardHover = {() => {}}
      history={history}
    />,
    {
      createNodeMock: () => {
        return {};
      }
    })
    .toJSON();

  expect(tree).toMatchSnapshot();
});
