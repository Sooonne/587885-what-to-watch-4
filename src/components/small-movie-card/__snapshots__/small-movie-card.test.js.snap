// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should SmallMovieCard render correctly 1`] = `
<article
  className="small-movie-card catalog_movies-card"
  onClick={[Function]}
  onMouseOut={[Function]}
  onMouseOver={[Function]}
>
  <div
    className="small-movie-card__image"
  >
    <video
      height="175"
      poster="/img/the-grand-budapest-hotel-poster.jpg"
      src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
      width="280"
    />
  </div>
  <h3
    className="small-movie-card__title"
  >
    <a
      className="small-movie-card__link"
      href="movie-page.html"
    >
      Bohemian Rhapsody
    </a>
  </h3>
</article>
`;
